libminifi/include/utils/ThreadPool.h

```diff
@@ -227,11 +227,10 @@ class ThreadPool {
    * Returns true if a task is running.
    */
   bool isTaskRunning(const std::string &identifier) const {
-    try {
-      return task_status_.at(identifier) == true;
-    } catch (const std::out_of_range &) {
+    const auto iter = task_status_.find(identifier);
```

Comment:
I know it's old code but I wonder why aren't we locking the `worker_queue_mutex_` or the `manager_mutex_` mutex here, even though we lock at least one of them every other place we access the `task_status_` map

Reply:
Good question, I added locking `worker_queue_mutex_` as it is locked by most other locations of accessing `task_status_`, and its documentation comment says `mutex to  protect task status and delayed queue`.